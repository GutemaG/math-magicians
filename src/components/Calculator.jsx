import React, { useState } from 'react';
import calculate from '../logic/calculate';
import DisplayButtons from './DisplayButtons';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });

  const handleEvent = (event) => {
    const value = event.target.innerText;
    const res = calculate(state, value);
    setState(res);
  };
  const { total, next, operation } = state;
  const r = (total || '') + (operation || '') + (next || '') || 0;
  return (
    <div className="calculator">
      <h1>Lets Do Some Math!</h1>
      <div>
        <input className="calculator-input" disabled value={r} />
        <DisplayButtons handleClick={handleEvent} />
      </div>
    </div>
  );
};

export default Calculator;
