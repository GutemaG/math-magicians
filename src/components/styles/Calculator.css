.btns {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  width: 550px;
  cursor: pointer;
}

.calculator {
  display: flex;
  justify-content: space-around;
  max-width: 80%;
  margin: auto;
}

.calculator-input {
  width: 550px;
  padding: 20px;
  background-color: #ccc;
  border: none;
  text-align: right;
}

.calculator-input:focus {
  border: none;
  outline: none;
}

.btn {
  border: 1px solid #ccc;
  padding: 20px;
  text-align: center;
  background-color: #fff;
  cursor: pointer;
}

.operator {
  background-color: orange;
}

.zero {
  grid-column: 1/ span 2;
}
