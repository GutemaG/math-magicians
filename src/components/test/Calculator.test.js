import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Testing Calculator component', () => {
  it('Calculator component rendered', () => {
    const tree = renderer.create(
      <Calculator />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing calculator component user interaction by click button', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('1'));
    expect(screen.getByText('1')).toBeInTheDocument();
  });

  it('Testing calculator component user interaction for adding two numbers', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('8'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTestId('resultInput')).toHaveValue('15');
  });

  it('Testing calculator component user interaction for multiplying two numbers', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTestId('resultInput')).toHaveValue('81');
  });
});
