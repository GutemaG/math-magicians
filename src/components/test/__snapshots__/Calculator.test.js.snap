// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator component Calculator component rendered 1`] = `
<div
  className="calculator"
>
  <h1>
    Lets Do Some Math!
  </h1>
  <div>
    <input
      className="calculator-input"
      data-testid="resultInput"
      disabled={true}
      value={0}
    />
    <div
      className="btns"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
